pkg/secrethub/credential.go

```diff
@@ -94,7 +94,7 @@ func NewCredential(credential string, passphrase string) (Credential, error) {
 // CredentialDecoder converts a payload into a Credential.
 type CredentialDecoder interface {
 	// Decode decodes a payload into a Credential.
-	Decode(payload []byte) (Credential, error)
+	Decode(payload []byte) (*RSACredential, error)
```

Comment:
If this is all RSA specific, should we move it to a separate file or rename this one?